src/RemoteCall/DeleteInstanceTrait.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Staffbase\plugins\sdk\RemoteCall;
+
+trait DeleteInstanceTrait
+{
+    /**
+     * Exit the script
+     *
+     * if a remote call was not handled by the user we die hard here
+     */
+    protected function exitRemoteCall(): void
+    {
+        error_log("Warning: The exit procedure for a remote call was not properly handled.");
+        exit;
+    }
+
+    private function deleteInstance(string $instanceId, RemoteCallInterface $remoteCallHandler): void
+    {
+        if ($remoteCallHandler instanceof DeleteInstanceCallHandlerInterface) {
+            $result = $remoteCallHandler->deleteInstance($instanceId);
+        } else {
+            // we will accept unhandled calls with a warning
+            $result = true;
+            error_log("Warning: An instance deletion call for instance $instanceId was not handled.");
+        }
+
+        // finish the remote call
+        if ($result) {
+            $remoteCallHandler->exitSuccess();
+        } else {
+            $remoteCallHandler->exitFailure();
+        }
+
+        $this->exitRemoteCall();
```

Comment:
I also stumbled upon this, and wasn't sure if there could be a case, where this will be reached. But I think you're right.

Reply:
There might be a case, where an extended class of `AbstractRemoteCallHandler` overload the methods without exiting. 
Maybe you should leave it as it is. (if it's copied)

Reply:
Now I know why this is there. The `$remoteCallHandler` only needs to be implementing the `RemoteCallInterface` but if the implementation doesn't include an exit call, it will land in the last line and then logs an error exits.